packages/blob-storage-connector-azure/src/azureBlobStorageConnector.ts

```
@@ -0,0 +1,196 @@
+		Guards.uint8Array(this.CLASS_NAME, nameof(blob), blob);
+
+		try {
+			const id = Converter.bytesToHex(Sha256.sum256(blob));
+			const blockBlobClient: BlockBlobClient = this._azureContainerClient.getBlockBlobClient(id);
+
+			const readableStream = new Readable({
+				/**
+				 * Reads data from the blob and pushes it into the stream.
+				 * This method is automatically called when the stream is read from.
+				 * It pushes the blob data and signals the end of the stream.
+				 */
+				read() {
+					this.push(blob);
+					this.push(null);
+				}
+			});
+			await blockBlobClient.uploadStream(readableStream);
+
+			return `blob:${new Urn(AzureBlobStorageConnector.NAMESPACE, id).toString()}`;
+		} catch (err) {
+			throw new GeneralError(this.CLASS_NAME, "setBlobFailed", undefined, err);
+		}
+	}
+
+	/**
+	 * Get the blob.
+	 * @param id The id of the blob to get in urn format.
+	 * @returns The data for the blob if it can be found or undefined.
+	 */
+	public async get(id: string): Promise<Uint8Array | undefined> {
+		Urn.guard(this.CLASS_NAME, nameof(id), id);
+		const urnParsed = Urn.fromValidString(id);
+
+		if (urnParsed.namespaceMethod() !== AzureBlobStorageConnector.NAMESPACE) {
+			throw new GeneralError(this.CLASS_NAME, "namespaceMismatch", {
+				namespace: AzureBlobStorageConnector.NAMESPACE,
+				id
+			});
+		}
+
+		try {
+			const key = urnParsed.namespaceSpecific(1);
+			const blobClient: BlobClient = this._azureContainerClient.getBlobClient(key);
+			const downloadResponse = await blobClient.download();
+
+			if (!downloadResponse.errorCode && downloadResponse?.readableStreamBody) {
+				const readableStream = downloadResponse.readableStreamBody as Readable;
+
+				const chunks: Uint8Array[] = [];
+				for await (const chunk of readableStream) {
+					chunks.push(chunk);
+				}
+
+				const buffer = Buffer.concat(chunks);
+				return new Uint8Array(buffer);
+			}
+			return undefined;
+		} catch {}
```

Probably need to throw an error in here